### PR TITLE
Fix #1856: Deletion of stack

### LINKED ISSUE #1856: Deletion of first stack note deletes also the child notes

### PR DESCRIPTION
**Pull request by Benjosh95**
## Description
Bug: Deletion of parent-note deleted the entire stack automatically.
Fix: Different note-deletion-actions get handled via confirmation-dialogs. 
Style: Changed confirmation-dialogs according to styling (https://xd.adobe.com/view/4a92e55c-64fc-401f-a671-0c565d51d563-1fc1/)

## Checklist
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] The light- and dark-theme are both supported and tested
- [x] The design was implemented and is responsive for all devices and screen sizes
- [x] The application was tested in the most commonly used browsers (e.g. Chrome, Firefox, Safari)

## Visual Changes
**The following Demo shows:**

2 Deletion-dialogs and its functionality:
- Delete parent-note and delete stack 
- Delete note

Changed design of confirmation-dialog/s:
- Return-to-homepage-dialog
- Delete-board-dialog

Dark-mode and the responsive design

https://user-images.githubusercontent.com/49522775/228359902-ac8059cd-1084-4663-845f-183c4b98d002.mp4

### CODE REVIEW

File: src/components/ConfirmationDialog/ConfirmationDialog.scss [hunk 1]
```diff
@@ -14,20 +14,79 @@
...
+.confirmation-dialog__warning > svg {
+  color: var(--accent-color);
+  min-width: 14px;
+  max-width: 18px;
+  opacity: 0.8;
 }
+.confirmation-dialog__warning > p {
+  margin: 0px;
+  font-size: 16px;
+  letter-spacing: 0.25px;
```
Comment (Resaki1) [inline src/components/ConfirmationDialog/ConfirmationDialog.scss:69]:
```suggestion
  letter-spacing: $letter-spacing--small;
```

File: src/components/ConfirmationDialog/ConfirmationDialog.scss [hunk 2]
```diff
@@ -14,20 +14,79 @@
...
+}
+
+@media screen and (max-width: 450px) {
```
Comment (Resaki1) [inline src/components/ConfirmationDialog/ConfirmationDialog.scss:72]:
```suggestion
@media #{$mini-smartphone} {
```

File: src/components/ConfirmationDialog/ConfirmationDialog.scss [hunk 3]
```diff
@@ -65,7 +124,8 @@
   height: $icon--extralarge;
   width: $icon--extralarge;
   border-radius: $rounded--full;
-  border: 2px solid $color-backlog-blue;
+  border: 2px transparent;
+  box-shadow: 0 16px 32px 0 rgb(0 87 255 / 24%);
```
Comment (Resaki1) [inline src/components/ConfirmationDialog/ConfirmationDialog.scss:128]:
```suggestion
  box-shadow: 0 16px 32px 0 rgba($color-backlog-blue, 0.24);
```

File: src/components/ConfirmationDialog/ConfirmationDialog.scss [hunk 4]
```diff
@@ -88,6 +148,7 @@
     color: $color-white;
     background-color: $color-dark-one;
     border-color: $color-planning-pink;
+    box-shadow: 0 16px 32px 0 rgba(0, 0, 0, 0.2);
```
Comment (Resaki1) [inline src/components/ConfirmationDialog/ConfirmationDialog.scss:151]:
```suggestion
    box-shadow: $box-shadow--dark;
```

File: src/components/ConfirmationDialog/ConfirmationDialog.scss [hunk 5]
```diff
@@ -106,8 +167,9 @@
 
   .confirmation-dialog__close-button {
     color: $color-planning-pink;
-    background-color: $color-dark-one;
+    background-color: $color-dark-mode;
     border-color: $color-planning-pink;
+    box-shadow: 0 16px 32px 0 rgba(0, 0, 0, 0.2);
```
Comment (Resaki1) [inline src/components/ConfirmationDialog/ConfirmationDialog.scss:172]:
```suggestion
    box-shadow: $box-shadow--dark;
```

**Review by Resaki1 — CHANGES_REQUESTED** [review]

**Comment by Resaki1** [comment]
@brandstetterm @BenedictHomuth could one of you review the backend code?

**Comment by Benjosh95** [comment]
@Resaki1 Thank you for the review :)! Fixed the points as u suggested.

**Review by Resaki1 — APPROVED** [review]

File: server/src/database/notes.go [hunk 6]
```diff
@@ -316,36 +317,68 @@ func (d *Database) DeleteNote(caller, board, id uuid.UUID) error {
 
 	if precondition.Author == caller || precondition.CallerRole == types.SessionRoleModerator || precondition.CallerRole == types.SessionRoleOwner {
 		previous := d.db.NewSelect().Model((*Note)(nil)).Where("id = ?", id).Where("board = ?", board)
+
+		updateBoard := d.db.NewUpdate().
+			Model((*Board)(nil)).
+			Set("shared_note = null").
+			Where("id = ? AND shared_note = ?", board, id)
```
[the remark this answers is not in the document]
Comment (Benjosh95) [reply server/src/database/notes.go:324]: No, there is no specific reason. I guess i liked it more in that order :D

File: server/src/database/notes_test.go [hunk 7]
```diff
@@ -482,10 +486,46 @@ func testDeleteSharedNote(t *testing.T) {
...
-	deleteNoteError := testDb.DeleteNote(author.ID, notesTestBoard.ID, noteC1.ID)
+	deleteNoteError := testDb.DeleteNote(author.ID, notesTestBoard.ID, deleteStack, noteC1.ID)
 	assert.Nil(t, deleteNoteError)
 
 	updatedBoard, getUpdatedBoardError := testDb.GetBoard(notesTestBoard.ID)
 	assert.Nil(t, getUpdatedBoardError)
 	assert.Equal(t, uuid.NullUUID{Valid: false}, updatedBoard.SharedNote)
 }
+
+func testDeleteStackParent(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (Benjosh95) [reply server/src/database/notes_test.go:497]: Thank you. I improved the test in this commit "fix updateRanks bug, improve tests"

**Pull request merged**